File: abc/abc196/a.java
// https://atcoder.jp/contests/abc196/submissions/43222452

import java.util.Scanner;

class Main {
  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    int a = stdIn.nextInt();
    int b = stdIn.nextInt();
    int c = stdIn.nextInt();
    int d = stdIn.nextInt();

    int ans = -10009;
    for(int x = a; x <= b; x++) {
      for(int y = c; y <= d; y++) {
        if(ans < x - y) {
          ans = x - y;
        }
      }
    }

    System.out.println(ans);
  }
}

File: abc/abc196/a.cs
// https://atcoder.jp/contests/abc196/submissions/46508219

using System;

public class A
{
  static void Main()
  {
    string[] t = Console.ReadLine().Split(" ");
    int a = int.Parse(t[0]);
    int b = int.Parse(t[1]);
    t = Console.ReadLine().Split(" ");
    int c = int.Parse(t[0]);
    int d = int.Parse(t[1]);

    int ans = -1000;
    for(int i = a; i <= b; i++)
    {
      for(int j = c; j <= d; j++)
      {
        if(ans < i - j)
        {
          ans = i - j;
        }
      }
    }

    Console.Write(ans);
  }
}

